### PR TITLE
FilterParametersUpdater manages filters in a better way

### PR DESCRIPTION
**Pull request by jolelievre**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | 1.7.5.x
| Description?  | FilterParametersUpdater manages overriding between request, persisted and default parameters in list and catalog
| Type?         | bug fix
| Category?     | BO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | Fixes https://github.com/PrestaShop/PrestaShop/issues/10703 <br /> Fixes https://github.com/PrestaShop/PrestaShop/issues/10257
| How to test?  | Go to Catalog > Products, test the filters in the list you will see that they work, then go to a product page Click on the product list you will see the last list of products you had in the catalog

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/10728)

### CODE REVIEW

File: src/Adapter/Product/FilterParametersUpdater.php [hunk 1]
```diff
@@ -55,72 +55,49 @@ public function setPositionOrdering($filterParameters, $orderBy, $hasCategoryFil
...
-    }
+    private function getParameter(
+        $parameterName,
+        array $queryFilterParameters,
+        array $persistedFilterParameters,
+        array $defaultFilterParameters
+    ) {
+        if (!empty($queryFilterParameters) && isset($queryFilterParameters[$parameterName])) {
+            $value = $queryFilterParameters[$parameterName];
+        } else if (!empty($persistedFilterParameters) && isset($persistedFilterParameters[$parameterName])) {
```
Comment (PierreRambaud) [inline src/Adapter/Product/FilterParametersUpdater.php]: You can use only isset here, no need to have an empty check.
same above.

File: src/Adapter/Product/FilterParametersUpdater.php [hunk 2]
```diff
@@ -55,72 +55,49 @@ public function setPositionOrdering($filterParameters, $orderBy, $hasCategoryFil
...
+            $value = $persistedFilterParameters[$parameterName];
+        } else {
+            $value = $defaultFilterParameters[$parameterName];
```
Comment (PierreRambaud) [inline src/Adapter/Product/FilterParametersUpdater.php:102]: You must be sure `$defaultFilterParameters[$parameterName]` exists.

Comment (jolelievre) [reply src/Adapter/Product/FilterParametersUpdater.php:102]: Yeah I hesitated to let PHP throw an error.. but I'm gonna throw a nice Exception instead ^^

**Review by PierreRambaud — APPROVED** [review]

**Comment by mickaelandrieu** [comment]
Thanks !

**Comment by PierreRambaud** [comment]
Thanks @jolelievre

**Pull request merged**